### PR TITLE
Power_Supervisor: Add MCP39F521

### PR DESCRIPTION
**Pull request by antoniovazquezblanco**
http://ww1.microchip.com/downloads/en/DeviceDoc/20005442A.pdf

imagen

------------
Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [x] Provide a URL to a datasheet for the symbol(s) you are contributing
- [x] An example screenshot image is very helpful
- [x] Ensure that the associated footprints match the [official footprint library](https://github.com/kicad/kicad-footprints)
- [x] If there are matching footprint PRs, provide link(s) as appropriate
- [x] Check the output of the Travis automated check scripts - fix any errors as required

### CODE REVIEW

File: Power_Supervisor.lib [hunk 1]
```diff
@@ -140,6 +140,50 @@ X VSS 3 -100 -400 100 U 50 50 1 1 W
...
 #
+# MCP39F521
+#
+DEF MCP39F521 U 0 20 Y Y 1 F N
+F0 "U" -450 750 50 H V C CNN
+F1 "MCP39F521" 350 750 50 H V C CNN
+F2 "Package_DFN_QFN:QFN-28-1EP_5x5mm_P0.5mm_EP3.35x3.35mm" 0 100 50 H I C CNN
+F3 "" 0 100 50 H I C CNN
+$FPLIST
+ QFN*1EP*5x5mm*P0.5mm*EP3.35x3.35mm*
```
Comment (DanSGiesbrecht) [inline Power_Supervisor.lib]: I've seen other PRs recently omit the EP size from the filter, for robustness.
```suggestion
 QFN*1EP*5x5mm*P0.5mm*
```

File: Power_Supervisor.dcm [hunk 2]
```diff
@@ -162,6 +162,12 @@ K supervisor reset push-pull
 F http://ww1.microchip.com/downloads/en/DeviceDoc/11187f.pdf
 $ENDCMP
 #
+$CMP MCP39F521
+D I2C Power Monitor with Calculation and Energy Accumulation,
```
Comment (DanSGiesbrecht) [inline Power_Supervisor.dcm]: I think that you can remove the comma at the end.
```suggestion
D I2C Power Monitor with Calculation and Energy Accumulation
```

File: Power_Supervisor.lib [hunk 3]
```diff
@@ -140,6 +140,50 @@ X VSS 3 -100 -400 100 U 50 50 1 1 W
...
+X REFIN+/OUT 23 600 -400 100 L 50 50 1 1 B
+X DGND 24 100 -900 100 U 50 50 1 1 W
+X DVDD 25 100 800 100 D 50 50 1 1 W
+X ~MCLR 26 600 600 100 L 50 50 1 1 I
+X DGND 27 100 -900 100 U 50 50 1 1 P N
+X ~DR 28 600 -600 100 L 50 50 1 1 O
+X DGND 29 100 -900 100 U 50 50 1 1 P N
+X NC 3 -500 0 100 R 50 50 1 1 N N
+X COMMON_B 4 -600 -200 100 R 50 50 1 1 I
+X COMMON_A 5 -600 -100 100 R 50 50 1 1 O
```
Comment (DanSGiesbrecht) [inline Power_Supervisor.lib]: Shouldn't `COMMONA` and `COMMONB` be passive? I couldn't find a reference to them being output and input respectively.

File: Power_Supervisor.lib [hunk 4]
```diff
@@ -140,6 +140,50 @@ X VSS 3 -100 -400 100 U 50 50 1 1 W
...
+X OSCI 6 -600 -600 100 R 50 50 1 1 I
+X OSCO 7 -600 -700 100 R 50 50 1 1 I
```
Comment (DanSGiesbrecht) [inline Power_Supervisor.lib]: Should `OSCO` be passive?

File: Power_Supervisor.lib (hunk 1 above)
Comment (antoniovazquezblanco) [reply Power_Supervisor.lib]: Agree

**Comment by antoniovazquezblanco** [comment]
Agree with all the comments.

Just a thing. I am not sure this device should be in this lib. Do you think this is the apropriate place?

Thanks for your work!

**Comment by DanSGiesbrecht** [comment]
I think that this is the appropriate library. :)

One thing that I actually meant to edit in my comments: I think that a better description might be: `Power Monitor, Calculation and Energy Accumulation, I2C, 16-bit, QFN-28`.

**Comment by antoniovazquezblanco** [comment]
Done, thank you very much!

**Comment by antoniovazquezblanco** [comment]
Done! Sorry!

**Comment by DanSGiesbrecht** [comment]
All good! Looks great :100:

**Pull request merged**